SampleOS Initial commit

davidcassany opened the pull request:
This commit includes a rough example of a drivative root filesystem
based on top of cOS. It includes a dummy http service and few little
configurations that differ from the cOS.

davidcassany committed:
SampleOS Initial commit

This commit includes a rough example of a drivative root filesystem
based on top of cOS. It includes a dummy http service and few little
configurations that differ from the cOS.

Signed-off-by: David Cassany <dcassany@suse.com>

mudler commented on packages/sampleOS/02_upgrades.yaml:
```diff
@@ -26,25 +26,25 @@ stages:
          permissions: 0600
          owner: 0
          group: 0
-       environment_file: /etc/cos-upgrade-image
```
This will likely break upgrades, currently it's hardcoded here: https://github.com/rancher-sandbox/cOS-toolkit/blob/master/packages/installer/upgrade.sh#L62

mudler commented on Makefile:
```diff
@@ -0,0 +1,107 @@
+
+export REPO_CACHE
+ifneq ($(strip $(REPO_CACHE)),)
+	BUILD_ARGS+=--image-repository $(REPO_CACHE)
```
Just a note about the image reusal, luet 0.12.0 allows now to pull from different caches, see : https://github.com/mudler/luet/issues/185#issuecomment-815562222

davidcassany replied:
Yes I added the `--pull-repository` build flag in the Makefile :+1:

mudler commented on values/opensuse.yaml:
```diff
@@ -1,24 +0,0 @@
-image: opensuse/leap:15.2
```
I'd keep this around (maybe by changing its name) as an example on how to apply templates. It is also required in case someone wants to rebuilds off entirely all the images

davidcassany replied:
Yes I have been thinking about it, and I was wondering if it wouldn't be easier to allow `values` to be split in multiple files. So for instance we could combine the base `values` provided by cOS and then user add/override values with an extra yaml file. 

What I don't like of keeping it around is that then we are kind a forking the original and it could eventually cause hard to notice inconsistencies on cOS base upgrades. So if we keep the original from the submodule we know it is in sync with the cOS base and additional tweaks could be applied at package level (most likely the top level one).

mudler replied:
makes sense, I'll open an issue to allow multiple values file to be applied during build :+1: maybe we can just retain a values file for the local tweakings as an example, without copying the content from the original cOS values files

Edit: Issue created here https://github.com/mudler/luet/issues/198

davidcassany committed:
Make use of the `--pull-repository` luet flag

This commit enables the use of `--pull-repository` so we can pull
images from multiple registries and/or repositories at build time.

Signed-off-by: David Cassany <dcassany@suse.com>

davidcassany committed:
Adding a build action

This commit adds a the build, publish and iso workflows.
The build and iso happen at any push to any branch or pull request and
the publish is only executed on master branch.

Signed-off-by: David Cassany <dcassany@suse.com>

mudler commented on .github/sign-images:
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
```
For an example I don't think we need to include this, I assume that signed images it's something to configure later on, if someone wants really to (similarly, the default upgrade channel should be set with `verified: false`)

mudler commented on Makefile:
```diff
@@ -0,0 +1,113 @@
+BACKEND?=docker
```
After yesterday conversation, maybe better to move this to a single script, or either reuse the `.envrc` from `cos-toolkit`? wdyt?

davidcassany replied:
Probably we can try to simplify the Makefile parameters slightly, even it is already quite simple and additionally document how to run local build without the makefile using a containerized luet, a three steps commands (build, create repo and geniso).

mudler commented:
Overall looks good! Just few minor nitpicks from yesterday discussion that we might also address in follow-ups. ( For example, we need also to provide a short doc as well)

davidcassany commented:
> Overall looks good! Just few minor nitpicks from yesterday discussion that we might also address in follow-ups. ( For example, we need also to provide a short doc as well)

Yes I agree we should then run few iterations to polish docs and eventually some details on how to present/document the build and customization procedure.
